src/schema/v2/city/index.ts

```diff
@@ -234,7 +235,7 @@ const lookupCity = (slug: string, cities: TCity[]) => {
   const city = cities.find((c) => c.slug === slug)
 
   if (!city) {
-    throw new Error(`City "${slug}" not found`)
+    throw new HTTPError(`City "${slug}" not found`, 404)
```

Comment:
Didn't know we needed to use this to support `@principleField` 👍 

I'll add a todo about adding a note around here https://github.com/artsy/metaphysics/blob/master/src/lib/getPrincipalFieldDirectivePath.ts

Reply:
I'd also be helpful to co-locate all of this principle field stuff under a `directives/principleField` directory rather than spreading it throughout the codebase: 

<img width="528" alt="Screen Shot 2021-10-04 at 4 22 26 PM" src="https://user-images.githubusercontent.com/236943/135937435-a8095020-7856-45b7-90d2-76d9cee6bdc4.png">
:

Reply:
This isn't explicitly a principal field thing, I suppose. We only need to raise this type of error here because we're not getting a status code from the server since it's just being thrown in a hash lookup.